pkg/sql/row/fetcher.go

```diff
@@ -237,6 +237,10 @@ type Fetcher struct {
 	// correctness. It is set only during SCRUB commands.
 	isCheck bool
 
+	// IgnoreUnexpectedNulls allows Fetcher to return null values for non-nullable
+	// columns and is only used for debugging.
+	IgnoreUnexpectedNulls bool
```

Comment:
that's not quite true, you're currently setting it in error reporting code as well. We'll see if any tests complain about this